docs: better document auth scopes

ankddev opened the pull request:
Fixes #6092.
# Description
I have added new help topic `scopes`, which contains information [about auth scopes](https://docs.github.com/en/apps/oauth-apps/building-oauth-apps/scopes-for-oauth-apps#available-scopes). Also I have added mention of this help topic to `gh auth login` and `gh auth refresh`.
# Screenshots
## Help topic
![image](https://github.com/user-attachments/assets/d82b2ac3-d50f-4109-83a5-47bbb6317dfb)
## `gh auth login` command help
![image](https://github.com/user-attachments/assets/81e96cb3-7f04-41b2-87fd-5f57e06359be)
## `gh auth refresh` command help
![image](https://github.com/user-attachments/assets/4c38c192-e9b4-40dc-9184-e55a892b2163)

ankddev committed:
docs: add help topic for auth scopes

ankddev committed:
docs: add mention of scopes help topic in `auth login` command help

ankddev committed:
docs: add mention of scopes help topic in `auth refresh` command help

ankddev committed:
Merge branch 'cli:trunk' into better-document-auth-scopes

andyfeller commented:
@ankddev : wow, talk about a blast from the past! ✨ given the age on this issue, I appreciate your patience as I get up to speed on the original issue and the proposed changes. 🙇

andyfeller commented on pkg/cmd/auth/login/login.go:
```diff
@@ -84,6 +84,8 @@ func NewCmdLogin(f *cmdutil.Factory, runF func(*LoginOptions) error) *cobra.Comm
 			prompting to create and upload a new key if one is not found. This can be skipped with
 			%[1]s--skip-ssh-key%[1]s flag.
+
+			Run %[1]sgh help scopes%[1]s to see information about auth scopes.
```
```suggestion
			For more information on OAuth scopes, <https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps/>.
```

andyfeller commented on pkg/cmd/auth/refresh/refresh.go:
```diff
@@ -72,6 +72,8 @@ func NewCmdRefresh(f *cmdutil.Factory, runF func(*RefreshOptions) error) *cobra.
 			inactive account, you will have to use %[1]sgh auth switch%[1]s to that account first before using
 			this command, and then switch back when you are done.
+
+			Run %[1]sgh help scopes%[1]s to see information about auth scopes.
```
```suggestion
			For more information on OAuth scopes, <https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps/>.
```

andyfeller committed:
Update gh auth commands to point to GitHub Docs

After initial discussion with core maintainers, it didn't seem prudent to duplicate the GitHub Docs information on OAuth scopes within the GitHub CLI codebase.

Instead, this commit updates help usage information in relevant commands on the page within GitHub Docs for more information.

andyfeller reviewed (approved):
Thanks again @ankddev for the contribution! 🙇 

Though the original issue suggested having `gh` provide this information directly to users, I don't know that it makes sense today.  Instead, I think ensuring users know where to lookup the information especially as we cannot programmatically retrieve this from the GitHub API.  Rather than leaving suggestions for you to commit, I went ahead with the necessary changes to close this out.
